src/borg/helpers/passphrase.py

```diff
@@ -139,3 +141,23 @@ def __repr__(self):
 
     def kdf(self, salt, iterations, length):
         return pbkdf2_hmac('sha256', self.encode('utf-8'), salt, iterations, length)
+
+    def argon2(
+        self,
+        *,
+        salt: bytes,
+        time_cost,
+        memory_cost,
+        parallelism,
+        type: argon2.low_level.Type
+    ) -> (bytes, bytes):
+        key = argon2.low_level.hash_secret_raw(
```

Comment:
dinosaurs with laser guns! :-)

Reply:
The high level API of argon2 is designed for _password hashing_. Using it as a _key derivation function_ would be awkward and have no benefit as far as I can tell.

Reply:
yes, I've seen that the HL api is not usable for us.